tools/fill_conf_values.py

```diff
@@ -36,12 +36,35 @@ def hash_filter(string, htype):
     return h.hexdigest()
 
 
+def nginx_brotli_module_installed():
+    # we run the `nginx -V` command to check if the nginx server is compiled with brotli support
+    # if it is, we'll find "brotli" in the output
+    import subprocess
+
+    try:
+        output = subprocess.check_output(
+            ["nginx", "-V"], stderr=subprocess.STDOUT
+        ).decode("utf-8")
+        return "brotli" in output
+    except subprocess.CalledProcessError:
+        return False
+
+
+def is_macos():
+    return os.uname().sysname == "Darwin"
+
+
 custom_filters = {"md5sum": md5sum, "version": version, "hash": hash_filter}
 
+USE_BROTLI = nginx_brotli_module_installed()
+IS_MACOS = is_macos()
+
 
 def fill_config_file_values(template_paths):
     log("Compiling configuration templates")
     env, cfg = load_env()
+    cfg["server"]["use_brotli"] = USE_BROTLI
+    cfg["server"]["is_macos"] = IS_MACOS
```

Comment:
Probably not needed; just do all the logic for whether or not brotli should be loaded here.

Reply:
actually it is, because on mac once installed with homebrew brotli doesnt need to be dynamically loading in the nginx config. But if installed with linux it looks like you need to. So I do need to have both of these as separate arguments

Reply:
what I need to change is that I actually need to find the full path to the brotli module, which is trickier. But that way no need to figure out if its mac or not. I just need to find the path to where the nginx modules are stored and I'm done, I can always dynamically load brotli

Reply:
Yep, just got it to work on linux. I just need to verify that the dynamic import on macos is possible, and then I can have one single method for both.

Reply:
@stefanv I managed to install brotli both dynamically and statically in nginx while on linux. I think I've got it figured out. I'll add a piece of code which if statically installed, does not add the lines in the baselayer config to dynamically load brotli, and if installed dynamically, tries to figure out where the modules are so they can be loaded. If it can't figure out where the modules are but brotli is installed, just display a warning when running the app and start it without brotli